libsolidity/codegen/ir/IRGenerationContext.h

```diff
@@ -61,6 +62,18 @@ class IRGenerationContext
 
 	MultiUseYulFunctionCollector& functionCollector() { return m_functions; }
 
+	/// Adds a function to the function generation queue and returns the name of the function.
+	std::string enqueueFunctionForCodeGeneration(FunctionDefinition const& _function);
+
+	/// Resolves a virtual function call into the right definition and queues it for code generation
+	/// code generation. Returns the name of the queued function.
+	std::string enqueueVirtualFunctionForCodeGeneration(FunctionDefinition const& _functionDeclaration);
+
+	/// Pops one item from the function generation queue. Returns nullptr if the queue is empty.
```

Comment:
```suggestion
	/// Pops one item from the function generation queue. Throws if the queue is empty.
```

Reply:
Done:
```
/// Pops one item from the function generation queue. Must not be called if the queue is empty.
```

I did not want to say in the above that it specifically throws because trying to use this function when the queue is empty is just a bug and behavior in that case should be considered undefined.